bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v2/PageImpl.java

```diff
@@ -311,4 +329,55 @@ public String getMainContentSelector() {
         }
         return htmlPageItems;
     }
+
+    @Override
+    public @Nullable String getCanonicalLink() {
+        if (this.canonicalUrl == null) {
+            String canonicalUrl;
+            try {
+                SeoTags seoTags = resource.adaptTo(SeoTags.class);
+                canonicalUrl = seoTags != null ? seoTags.getCanonicalUrl() : null;
+            } catch (NoClassDefFoundError ex) {
+                canonicalUrl = null;
+            }
+            this.canonicalUrl = canonicalUrl != null
+                ? canonicalUrl
+                : pathProcessor.externalize(currentPage.getPath(), request) + LinkHandler.HTML_EXTENSION;
```

Comment:
The HTML extension is already added by the LinkHandler#getPageLinkURL method.

Reply:
Afaik the `DefaultPathProcessor` does not call the `LinkHandler`. The `SeoTags` implementation returns urls that include the `.html` extension, thats why it is added here in the fallback case as well.

Reply:
The LinkHandler should used instead of calling the path processor directly. The purpose of the LinkHandler is to have unified handler to generate all links in the core components project.